components/chat/chatBar/search/catalog.tsx

```diff
@@ -0,0 +1,592 @@
+    lastIndexRef.current = lastIndex;
+    useImperativeHandle(ref, () => ({
+      focus: () =>
+        lastIndexRef.current != null &&
+        focusElement(itemKey(lastIndexRef.current)),
+    }));
+
+    return (
+      <Card
+        className={
+          'absolute bottom-14 z-[1000] p-4 max-h-[710px] w-1/2 overflow-auto border-small border-default-200 dark:border-default-100'
+        }
+      >
+        <Listbox
+          aria-label={'catalog'}
+          variant={'flat'}
+          disallowEmptySelection
+          selectionMode={'single'}
+          shouldFocusWrap={true}
+          disabledKeys={equippedTools}
+          defaultSelectedKeys={equippedTools}
+          selectedKeys={equippedTools}
+          onSelectionChange={(selected) => {
+            if (selected === 'all') return;
+            setLoading(String(Array.from(selected).pop()) ?? null);
+          }}
+          onKeyDown={handleKeyDown}
+        >
+          {resultsWithIndexes.map(([category, tools]) => (
+            <ListboxSection
+              aria-label={`catalog-section-${category}`}
+              key={category}
+              title={category}
+              showDivider={true}
+            >
+              {tools.map((tool) => {
+                return (
+                  <ListboxItem
+                    aria-label={itemKey(tool.index)}
+                    startContent={
+                      loading === tool.url ? <Spinner size="sm" /> : tool.icon
+                    }
+                    isReadOnly={loading !== null}
+                    id={itemKey(tool.index)}
+                    key={tool.url} // Using tool URL as the unique key
+                    value={tool.url}
+                    onFocus={() => setFocusedItem(tool.index)}
+                  >
+                    <Tooltip
+                      content={tool.description}
+                      placement="right"
+                      isOpen={focusedItem === tool.index}
+                      onOpenChange={(open) =>
+                        setFocusedItem(open ? tool.index : null)
+                      }
+                      closeDelay={0.5}
+                      classNames={{
+                        content: 'max-w-[250px]',
+                      }}
+                    >
+                      {tool.name}
+                    </Tooltip>
+                  </ListboxItem>
+                );
+              })}
+            </ListboxSection>
+          ))}
+        </Listbox>
+      </Card>
+    );
+  }
+);
+
+function focusElement(id: string) {
+  const element = document.getElementById(id);
+  if (!element) return;
+  element.focus();
+}
+
+type FeaturedTool = {
+  name: string;
+  description: string;
+  url: string;
+  tags: string[];
+  icon: JSX.Element;
+};
+type FeaturedTools = Record<string, FeaturedTool[]>;
+
+// note(tylerslaton) - This will eventually be retrieved from the tools site, for now we must endure the pain of hardcoding.
+const featuredTools: FeaturedTools = {
```

Comment:
we probably want to import this from `edit/configure/imports/toolCatalog.tsx` (or vice versa) instead of duplicating it in both places.